app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+
+
+def cache(func: callable) -> callable:
+    dictionary = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> any :
+        if args in dictionary:
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+            dictionary[*args] = func(*args)
```

Comment:
Do not perform unpacking when you assign a dit key with `args`, `args` are already placed in a tuple and you would like to store the tuple as a dict key.